src/btree/node.ts

```diff
@@ -142,10 +142,12 @@ abstract class NodeImpl<Value extends Hash | ReadonlyJSONValue> {
   entries: Array<Entry<Value>>;
   hash: Hash;
   abstract readonly level: number;
+  readonly isMutable: boolean;
 
-  constructor(entries: Array<Entry<Value>>, hash: Hash) {
+  constructor(entries: Array<Entry<Value>>, hash: Hash, isMutable: boolean) {
```

Comment:
With boolean params like this I often use named params, or a two value enum (e.g. Mutability.Mutable, Mutability.Immutable).  I've seen too many bugs due to folks inverting the params meaning in their head between looking at the function signature, and calling the function.  With named params the meaning of the param is obvious at the place where it is called.

Reply:
I agree, but I've been vary of using objects to reduce GC churn.

VSCode's `"editor.inlayHints.enabled": true` makes this less of a problem:

<img width="543" alt="Screen Shot 2021-11-17 at 10 14 23" src="https://user-images.githubusercontent.com/45845/142258431-1d0890d8-aba5-4833-b4f4-f3d012870864.png">

I'm fine changing to a const enum.